app/wait_time/location.py

```diff
@@ -0,0 +1,54 @@
+from datetime import datetime, timezone
+from typing import Dict, Any
+from app.common import BadDataError
+import json
+
+
+class UserLocation:
+    def __init__(
+        self,
+        aid: str,
+        latitude: float,
+        longitude: float,
+        timestamp: datetime = datetime.now(timezone.utc),
+    ):
+        self.aid = aid
+        self.latitude = latitude
+        self.longitude = longitude
+        self.timestamp = timestamp
+
+    @staticmethod
+    def from_dict(aid: str, dict: Dict[str, Any]) -> "UserLocation":
+        """
+        Creates a new UserLocation object from a Python Dictionary
+
+        :param dict: Dictionary of key-value pairs corresponding to location.
+        :returns: UserLocation from specified data
+        :raises BadDataError: If required data is missing from the dictionary
+        """
+        try:
+            return UserLocation(
+                aid,
+                dict["latitude"],
+                dict["longitude"],
+                dict.get("timestamp", datetime.now(timezone.utc)),
```

Comment:
You can specify a timestamp if you're parsing directly from firebase but if you want to create a new UserLocation from just a longitude and latitude you have the option to omit this, and it will default to now

Reply:
Ah sorry I read this wrong. I don't think the type conversion works though from the firebase type to datetime. You might have to do the same parse as in events.

Reply:
It looks like the DatetimeWithNanoSeconds inherits from datetime already, so we should be able to treat it as a datetime (tested by converting a DatetimeWithNanoseconds to ISO format string in the same way you would do to a datetime). The only difference is an additional nanosecond attribute. (for reference: https://googleapis.dev/python/google-api-core/latest/helpers.html#google.api_core.datetime_helpers.DatetimeWithNanoseconds)

Reply:
Interesting, then maybe my events code was unnecessary. If we can convert seamlessly then all good.